### PR TITLE
chore: Allow flake8 6.x

### LINKED ISSUE #281: Compat with flake8 >= 6
"SolverProblemError

Because flake8-broken-line (0.6.0) depends on flake8 (>=3.5,<6) and no versions of flake8-broken-line match >0.6.0,<0.7.0, flake8-broken-line (>=0.6.0,<0.7.0) requires flake8 (>=3.5,<6). So, because project depends on both flake8 (^6.0.0) and flake8-broken-line (^0.6.0), version solving failed."

### PR DESCRIPTION
**Pull request by jashparekh**
`flake8` 6.0 was recently released: https://flake8.pycqa.org/en/latest/release-notes/6.0.0.html.
No breaking change should impact the plugin.

I've kept an upper bound to be consistent with previous changes, but would you consider not setting one? [iscinumpy.dev/post/bound-version-constraints](https://iscinumpy.dev/post/bound-version-constraints/) is an interesting read on the subject.

Should resolve #281

### CODE REVIEW

**Comment by sobolevn** [comment]
Looks like we would need to drop 3.7 support for this to work 😒

**Comment by jashparekh** [comment]
> Looks like we would need to drop 3.7 support for this to work 😒

I can do that too

File: pyproject.toml
Comment (jashparekh) [inline pyproject.toml:43]: Minimum version required by `flake8`

**Comment by sobolevn** [comment]
I really don't know what to do.

1. I won't drop python3.7 in my projects until its EOL is reached
2. I still want users to be able to use recent flake8 releases

**Comment by jashparekh** [comment]
> I really don't know what to do.
> 
> 1. I won't drop python3.7 in my projects until its EOL is reached
> 2. I still want users to be able to use recent flake8 releases

I am not if there is a way out here as well, most of my apps are on `3.10` but I get the overhead that not everyone would be able to take on the overhead of migrating off of `3.7` just yet. I will read my up the release note closely later today. Not sure what was the reason for `flake8` to drop support for `3.7` so soon.

**Comment by sobolevn** [comment]
It is probably due to `importlib-metadata` problem

**Comment by jashparekh** [comment]
Ya that was exactly why `3.8.1` was preferred. Currently we are blocked with `flake8` upgrade due to this, I will hold off next week and let you make the final call here.

Happy Thanksgiving!

**Comment by jonyscathe** [comment]
Is there any issue with still allowing python 3.7?
That way those still on python 3.7 can use the latest flake8-broken-line with flake8==5.x.x and those on python >3.7 can start using flake8 6.

**Comment by sobolevn** [comment]
> That way those still on python 3.7 can use the latest flake8-broken-line with flake8==5.x.x and those on python >3.7 can start using flake8 6.

Sound like the best option for me.

**Comment by jashparekh** [comment]
> > That way those still on python 3.7 can use the latest flake8-broken-line with flake8==5.x.x and those on python >3.7 can start using flake8 6.
> 
> Sound like the best option for me.

Updated my PR accordingly!

**Comment by sobolevn** [comment]
Why do we need `typed-ast`?

**Comment by jashparekh** [comment]
> Why do we need `typed-ast`?

I could not figure the exact cause but if you look [here](https://github.com/wemake-services/flake8-broken-line/actions/runs/3549416516/jobs/5968715915#step:6:18), the tests for `python3.7` started failing because it was missing

**Comment by sobolevn** [comment]
Are you sure that it is a test-only dependency then?

**Comment by jashparekh** [comment]
> Are you sure that it is a test-only dependency then?

Can you approve my request to run the CI?

**Comment by jashparekh** [comment]
> Are you sure that it is a test-only dependency then?

yes

**Comment by jashparekh** [comment]
Hey @sobolevn - Is there something still blocking this PR? We are blocked across few projects with our Flake8 upgrade because of this

**Comment by bellini666** [comment]
If you can merge and do a new release, we are also blocked from updating flake8 until this gets released.

Also, I would suggest maybe removing the upper bound from the requirement so that this doesn't happen again in the future (maybe not in this PR to avoid taking more time to release it)

**Comment by jashparekh** [comment]
> If you can merge and do a new release, we are also blocked from updating flake8 until this gets released.
> 
> Also, I would suggest maybe removing the upper bound from the requirement so that this doesn't happen again in the future (maybe not in this PR to avoid taking more time to release it)

I like the idea of removing the upper bound but also agree that it should be a separate PR to do so. I can create the PR once this merged

**Comment by matt-carr** [comment]
> Looks like we would need to drop 3.7 support for this to work 😒

I don't think this is true? Someone correct me if I'm crazy here, but: 

* because this is a plugin, we count on the consumer's dependency resolution to pick versions
* There's nothing specifically in this library that is incompatible with py3.7
* If we expand the version clamp to flake8<=6, all that means is that we're stating that it can use any flake8 version between 3.5 and 6

So, basically, if we expand the version clamp, it's up to the user to pick if they want to stick with py3.7 and flake8 5.0.4 or bump to py3.8.1+ and flake8 6.0.0, but in either case this should be able to plug into either one because it's not dependent on any specific functionality in flake8 6 or python 3.8.1. 

Basically, _this plugin_ will have no compatibility problems, and so you should be able to _expand compatible versions_ to flake8 6

**Comment by jashparekh** [comment]
> > Looks like we would need to drop 3.7 support for this to work 😒
> 
> I don't think this is true? Someone correct me if I'm crazy here, but:
> 
> * because this is a plugin, we count on the consumer's dependency resolution to pick versions
> * There's nothing specifically in this library that is incompatible with py3.7
> * If we expand the version clamp to flake8<=6, all that means is that we're stating that it can use any flake8 version between 3.5 and 6
> 
> So, basically, if we expand the version clamp, it's up to the user to pick if they want to stick with py3.7 and flake8 5.0.4 or bump to py3.8.1+ and flake8 6.0.0, but in either case this should be able to plug into either one because it's not dependent on any specific functionality in flake8 6 or python 3.8.1.
> 
> Basically, _this plugin_ will have no compatibility problems, and so you should be able to _expand compatible versions_ to flake8 6

That is accurate, the current PR incorporates takes this into consideration and does not drop support for `3.7`

**Comment by jashparekh** [comment]
Hi @sobolevn - Following up on the review of this PR, since its blocking lot of people and projects

**Review by Dreamsorcerer — APPROVED** [review]
As mentioned above, this looks correct. Py 3.7 users will still install flake8 v5, while others users can install v6.

I'd probably make it a patch release though (0.6.1) given no code or backwards-compatibility changes have taken place.

**Comment by jashparekh** [comment]
> As mentioned above, this looks correct. Py 3.7 users will still install flake8 v5, while others users can install v6.
> 
> I'd probably make it a patch release though (0.6.1) given no code or backwards-compatibility changes have taken place.

Done

**Comment by jashparekh** [comment]
Hey @sobolevn - Reaching out one more time for a review on this PR

**Comment by decaz** [comment]
@sobolevn are any updates on this?

**Comment by jashparekh** [comment]
Hey @sobolevn - Reaching out for review again

**Comment by jashparekh** [comment]
Hi @sobolevn - Following up again on this, can you share what's holding this PR to be approved

**Comment by chopeen** [comment]
Hello @sobolevn - can you please let us know if and when this PR can be accepted?

If you need to abandon this plugin altogether, that's also valuable information - then we can consider forking the project to publish a new package to PyPI or look for alternatives.

**Comment by jashparekh** [comment]
Hi @sobolevn - I am planning to fork this project next week and publish a new package, please let us know soon if you plan to support this plugin

**Comment by chopeen** [comment]
FYI @jashparekh, I decided to switch my pipelines to Ruff https://beta.ruff.rs/docs/rules/.

**Comment by terencehonles** [comment]
Looks like this is superseded by #307

**Pull request closed without merging**
